File: php/src/Location.php
<?php

declare(strict_types=1);

namespace Cucumber\Gherkin;

/**
 * @psalm-api
 */
final class Location
{
    public function __construct(
        public readonly int $line,
        public readonly int $column,
    ) {
    }

    public function getColumn(): int
    {
        return $this->column;
    }
}

File: dart/lib/src/language/location.dart
class Location {
  const Location(this.line, this.column);

  static const int _unset = -1;

  static const empty = Location(_unset, _unset);

  final int line;

  final int column;

  bool get isEmpty => line == empty.line && column == empty.column;

  bool get hasColumn => column != _unset;
}
